client/client.go

```diff
@@ -86,20 +91,25 @@ func (c *MeshStackProviderClient) login() error {
 		return err
 	}
 
-	formData := url.Values{}
-	formData.Set("client_id", c.apiKey)
-	formData.Set("client_secret", c.apiSecret)
-	formData.Set("grant_type", "client_credentials")
+	loginRequest := loginRequest{
+		ClientId:     c.apiKey,
+		ClientSecret: c.apiSecret,
+	}
+
+	payload, err := json.Marshal(loginRequest)
+	if err != nil {
+		return err
+	}
 
-	req, _ := http.NewRequest(http.MethodPost, loginPath, strings.NewReader(formData.Encode()))
-	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	req, _ := http.NewRequest(http.MethodPost, loginPath, bytes.NewBuffer(payload))
+	req.Header.Add("Content-Type", "application/json")
 
 	res, err := c.httpClient.Do(req)
 
 	if err != nil {
 		return err
 	} else if res.StatusCode != 200 {
-		return errors.New(ERROR_AUTHENTICATION_FAILURE)
+		return errors.New(fmt.Sprintf("Status %d: %s", res.StatusCode, ERROR_AUTHENTICATION_FAILURE))
```

Comment:
d: we're still using `ERROR_AUTHENTICATION_FAILURE` for all status codes other than 200?